src/components/AppPosts.tsx

```diff
@@ -0,0 +1,128 @@
+import classNames from 'classnames';
+
+import { PostsList } from './PostList/PostsList';
+import { PostDetails } from './PostDetails/PostDetails';
+import { UserSelector } from './UserSelector';
+import { Loader } from './Loader';
+import { usePosts } from '../hooks/usePosts';
+import { ErrorMessages } from '../types/ErrorMessages';
+
+export const AppPosts = () => {
+  const {
+    posts,
+    comments,
+    isLoadingPosts,
+    isLoadingAdd,
+    isOpenDropdown,
+    isOpenSidebar,
+    isOpenCommentForm,
+    selectedUser,
+    selectedPost,
+    handleSelectUser,
+    handleDeleteComment,
+    handleInputName,
+    handleInputEmail,
+    handleInputMessage,
+    toggleDropdown,
+    toggleSidebar,
+    toggleCommentForm,
+    users,
+    isLoadingComments,
+    inputName,
+    inputEmail,
+    inputMessage,
+    onResetForm,
+    handleAddComment,
+    dropdownRef,
+    currentError,
+  } = usePosts();
```

Comment:
In this case, a better realization uses context to avoid props drilling. 
Now you move all methods to the hook. This encapsulates some logic, but don't change props usage.

I think you need try to impove this solution:
Try to think what you can unit/split?
How mede this list of property shorter?
Did you need all of this property? 
What you can use inside component, what minimal setup for hook? 
Can i use context in this case?